### PR TITLE
Update the developers guidelines

### PR DESCRIPTION
**Pull request by DelazJ**
Include the diff texts from the [dev guidelines on QGIS-website](https://qgis.org/en/site/getinvolved/development/qgisdevelopersguide/index.html)
Hope that i caught them all.

Ticket: refs #1566 and https://github.com/qgis/QGIS-Website/pull/406

- [x] Backport to LTR documentation is required

### Minimal requirements for merging *(for maintainers)*

- [ ] The description of this PR is coherent with the manual and does not provide wrong information.
- [ ] This PR passes the checks.

### CODE REVIEW

File: source/docs/developers_guide/codingstandards.rst [hunk 1]
```diff
@@ -294,6 +294,20 @@ e.g.:
...
+Comments
+========
+
+Comments to class methods should use a third person indicative style instead
+of the imperative style:
+
+.. code-block:: cpp
+
+    /**
+     * Creates a new QgsFeatureFilterModel, optionally specifying a \a parent.
```
Comment (havatv) [inline source/docs/developers_guide/codingstandards.rst]:
```suggestion
     * Creates a new QgsFeatureFilterModel, optionally specifying a parent.
```

Comment (havatv) [reply source/docs/developers_guide/codingstandards.rst]: Not sure what the \a was for, so I used a suggestion.

Comment (DelazJ) [reply source/docs/developers_guide/codingstandards.rst]: This is on purpose. Not sure of what it's but source code is full of that. seems it helps to reference attributes of the function and their display (see https://qgis.org/pyqgis/master/core/QgsFeatureFilterModel.html)

Comment (havatv) [reply source/docs/developers_guide/codingstandards.rst]: OK. Thanks for the information!

**Pull request merged**